Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -904,6 +906,10 @@ protected async Task RegisterResources(IServiceCollection serviceCollection)
 
             serviceCollection.AddSingleton(typeof(IResourceFileManager), typeof(ResourceFileManager));
 
+            AttachmentExtractor = new MediaBrowser.MediaEncoding.Attachments.AttachmentExtractor(LibraryManager, LoggerFactory, ApplicationPaths, FileSystemManager, MediaEncoder, MediaSourceManager, ProcessFactory);
+
+            serviceCollection.AddSingleton(AttachmentExtractor);
```

Comment:
You could just do this if all of these are in the `ServiceCollection`. No need for the property.
```suggestion
            serviceCollection.AddSingleton<MediaBrowser.MediaEncoding.Attachments.AttachmentExtractor>();
```

Reply:
I had to revert this one, it causes the attachment service to fail.

[2019-11-06 07:37:07.010 -05:00] [ERR] Error creating "MediaBrowser.Api.Attachments.AttachmentService"
System.InvalidOperationException: Unable to resolve service for type 'MediaBrowser.Controller.MediaEncoding.IAttachmentExtractor' while attempting to activate 'MediaBrowser.Api.Attachments.AttachmentService'.
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.ConstructorMatcher.CreateInstance(IServiceProvider provider)
   at Microsoft.Extensions.DependencyInjection.ActivatorUtilities.CreateInstance(IServiceProvider provider, Type instanceType, Object[] parameters)
   at Emby.Server.Implementations.ApplicationHost.CreateInstanceSafe(Type type) in /home/dev/jellyfin/jellyfin/Emby.Server.Implementations/ApplicationHost.cs:line 482

Reply:
fixed